platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/merkle/VirtualRootNode.java

```diff
@@ -880,69 +900,74 @@ public V replace(final K key, final V value) {
      * @return The removed value. May return null if there was no value to remove or if the value was null.
      */
     public V remove(final K key) {
+        assert currentModifyingThreadRef.compareAndSet(null, Thread.currentThread());
         throwIfImmutable();
         Objects.requireNonNull(key);
 
-        // Verify whether the current leaf exists. If not, we can just return null.
-        VirtualLeafRecord<K, V> leafToDelete = records.findLeafRecord(key, true);
-        if (leafToDelete == null) {
-            return null;
-        }
+        try {
```

Comment:
Minor: it would look more consistent if the assert at line 903 is right above the opening `try {`, just like you do in other places in this fix. This also builds a clear mental connection with the subsequent `finally{}` block where the atomicRef is cleared.

Reply:
Fair enough